### ISSUE TITLE
`@stylistic/selector-attribute-brackets-space-inside` writes only one of the two spellings of a selector carrying an inline comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#190]
Found while fixing [#178](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/178), whose review swept the rules that write a selector. This one predates that branch and reproduces without it.

### What minimal example or steps are needed to reproduce the bug?

```scss
.a // c
:not( [ t='y' ] ) {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-scss",
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/selector-attribute-brackets-space-inside": "never"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.scss" --config myconfig.json --fix
```

### Which Stylelint-related dependencies are you using?

```json
{
  "@stylistic/stylelint-plugin": "5.3.0",
  "stylelint": "17.14.1"
}
```

### What did you expect to happen?

```scss
.a // c
:not( [t='y'] ) {}
```

### What actually happened?

The file comes out exactly as it went in, and no warning survives to say so — `--fix` reports a clean pass on a file it never changed, which is the failure mode of [#126](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/126).

Add `selector-pseudo-class-parentheses-space-inside: "never"` to the configuration and the miss becomes corruption:

```scss
.a // ca // c
:not( [t='y'] ) {}
```

— the head of the selector is doubled, and the doubling grows with every further run of `--fix`.

### Do you have a proposal to fix the bug?

`postcss-scss` keeps two spellings of a selector carrying an inline comment: the raw, where every `//` comment is rewritten as a block one, and `raws.selector.scss`, the file's own spelling, which is the copy the stringifier prints. The rule reads the raw and writes its fix back to the raw alone:

```js
let selector = ruleNode.raws.selector ? ruleNode.raws.selector.raw : ruleNode.selector
…
if (ruleNode.raws.selector) ruleNode.raws.selector.raw = fixedSelector
```

so the printed copy never learns of the fix — hence the silent miss — and the two copies are left out of step, which is what the corruption grows from: the next rule reads the pair expecting them to tell one story, and `findSelectorInlineComments` walks them side by side into a wrong alignment.

The write-back it needs is the one `selector-descendant-combinator-no-non-space`, `selector-pseudo-class-parentheses-space-inside` and (since #178) `string-quotes` share:

```js
if (selectorRaws.scss) selectorRaws.scss = restoreSelectorInlineComments(fixedSelector, inlineComments)
```

with the positions of its warnings going through `toSelectorSourceIndex` on the way, since today they are counted in the raw and so drift two characters per comment from the file.

Worth checking the same sweep over any other rule that writes `raws.selector.raw` without the `scss` half.
